opslevel/resource_opslevel_check_tool_usage.go

```diff
@@ -1,101 +1,260 @@
+	Category    types.String `tfsdk:"category"`
+	Description types.String `tfsdk:"description"`
+	Enabled     types.Bool   `tfsdk:"enabled"`
+	EnableOn    types.String `tfsdk:"enable_on"`
+	Filter      types.String `tfsdk:"filter"`
+	Id          types.String `tfsdk:"id"`
+	Level       types.String `tfsdk:"level"`
+	Name        types.String `tfsdk:"name"`
+	Notes       types.String `tfsdk:"notes"`
+	Owner       types.String `tfsdk:"owner"`
+	LastUpdated types.String `tfsdk:"last_updated"`
+
+	ToolCategory         types.String    `tfsdk:"tool_category"`
+	ToolNamePredicate    *PredicateModel `tfsdk:"tool_name_predicate"`
+	ToolUrlPredicate     *PredicateModel `tfsdk:"tool_url_predicate"`
+	EnvironmentPredicate *PredicateModel `tfsdk:"environment_predicate"`
+}
+
+func NewCheckToolUsageResourceModel(ctx context.Context, check opslevel.Check, planModel CheckToolUsageResourceModel) CheckToolUsageResourceModel {
+	var stateModel CheckToolUsageResourceModel
+
+	stateModel.Category = RequiredStringValue(string(check.Category.Id))
+	stateModel.Description = ComputedStringValue(check.Description)
+	if planModel.Enabled.IsNull() {
+		stateModel.Enabled = types.BoolValue(false)
+	} else {
+		stateModel.Enabled = OptionalBoolValue(&check.Enabled)
+	}
+	if planModel.EnableOn.IsNull() {
+		stateModel.EnableOn = types.StringNull()
+	} else {
+		// We pass through the plan value because of time formatting issue to ensure the state gets the exact value the customer specified
+		stateModel.EnableOn = planModel.EnableOn
+	}
+	stateModel.Filter = OptionalStringValue(string(check.Filter.Id))
+	stateModel.Id = ComputedStringValue(string(check.Id))
+	stateModel.Level = RequiredStringValue(string(check.Level.Id))
+	stateModel.Name = RequiredStringValue(check.Name)
+	stateModel.Notes = OptionalStringValue(check.Notes)
+	stateModel.Owner = OptionalStringValue(string(check.Owner.Team.Id))
+
+	stateModel.ToolCategory = RequiredStringValue(string(check.ToolCategory))
+	if check.ToolNamePredicate != nil {
+		stateModel.ToolNamePredicate = NewPredicateModel(*check.ToolNamePredicate)
+	}
+	if check.ToolUrlPredicate != nil {
+		stateModel.ToolUrlPredicate = NewPredicateModel(*check.ToolUrlPredicate)
+	}
+	if check.EnvironmentPredicate != nil {
+		stateModel.EnvironmentPredicate = NewPredicateModel(*check.EnvironmentPredicate)
+	}
+
+	return stateModel
+}
+
+func (r *CheckToolUsageResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_check_tool_usage"
+}
+
+func (r *CheckToolUsageResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Check Tool Usage Resource",
+
+		Attributes: CheckBaseAttributes(map[string]schema.Attribute{
+			"tool_category": schema.StringAttribute{
+				Description: "",
```

Comment:
```suggestion
				Description: "The category that the tool belongs to.",
```
From [docs](https://registry.terraform.io/providers/OpsLevel/opslevel/latest/docs/resources/check_tool_usage)